tests/test_benchmark.py

```diff
@@ -120,20 +124,153 @@ def process(filename):
   def run(self, args):
     return run_js(self.filename, engine=self.engine, args=args, stderr=PIPE, full_output=True, assert_returncode=None)
 
+class CheerpBenchmarker(Benchmarker):
+  def __init__(self, name, engine, args=['-O3'], binaryen_opts=[]):
+    self.name = name
+    self.engine = engine
+    self.args = args
+    self.binaryen_opts = binaryen_opts[:]
+
+  def build(self, parent, filename, args, shared_args, emcc_args, native_args, native_exec, lib_builder, has_output_parser):
+    suffix = filename.split('.')[-1]
+    cheerp_temp = filename + '.cheerp.' + suffix
+    code = open(filename).read()
+    if 'int main()' in code:
+      main_args = ''
+    else:
+      main_args = 'argc, (%(const)s char**)argv' % {
+        'const': 'const' if 'const char *argv' in code else ''
+      }
+    open(cheerp_temp, 'w').write('''
+      %(code)s
+      void webMain() {
+        // TODO: how to read from commandline?
+        volatile int argc = 2;
+        typedef char** charStarStar;
+        volatile charStarStar argv;
+        argv[0] = "./cheerp.exe";
+        argv[1] = "%(arg)s";
+        main(%(main_args)s);
+      }
+    ''' % {
+      'arg': args[-1],
+      'code': code,
+      'main_args': main_args
+    })
+    cheerp_args = [
+      '-target', 'cheerp', '-cheerp-mode=wasm',
+    ]
+    self.parent = parent
+    if lib_builder:
+      # build as "native" (so no emcc env stuff), but with all the cheerp stuff
+      # set in the env
+      cheerp_args = cheerp_args + lib_builder(self.name, native=True, env_init={
+        'CC': '/opt/cheerp/bin/clang',
+        'CXX': '/opt/cheerp/bin/clang++',
+        'AR': '/opt/cheerp/bin/llvm-ar',
+        'LD': '/opt/cheerp/bin/clang',
+        'NM': '/opt/cheerp/bin/llvm-nm',
+        'LDSHARED': '/opt/cheerp/bin/clang',
+        'RANLIB': '/opt/cheerp/bin/llvm-ranlib',
+        'CFLAGS': '-target cheerp -cheerp-mode=wasm',
+        'CXXFLAGS': '-target cheerp -cheerp-mode=wasm',
+      })
+    final = os.path.dirname(filename) + os.path.sep + 'cheerp_' + self.name + ('_' if self.name else '') + os.path.basename(filename) + '.js'
+    final = final.replace('.cpp', '')
+    try_delete(final)
+    dirs_to_delete = []
+    try:
+      for arg in cheerp_args[:]:
+        if arg.endswith('.a'):
+          info = self.handle_static_lib(arg)
+          cheerp_args += info['files']
+          dirs_to_delete += [info['dir']]
+      cheerp_args = [arg for arg in cheerp_args if not arg.endswith('.a')]
+      #print(cheerp_args)
+      cmd = ['/opt/cheerp/bin/clang++'] + cheerp_args + [
+        '-cheerp-linear-heap-size=256',
+        '-cheerp-wasm-loader=' + final,
+        cheerp_temp,
+        '-Wno-writable-strings', # for how we set up webMain
+        '-o', final + '.wasm'
+      ] + shared_args
+      #print(' '.join(cmd))
+      subprocess.check_call(cmd)
+      self.filename = final
+      if self.binaryen_opts:
+        subprocess.check_call([
+          '/home/alon/Dev/2-binaryen/bin/wasm-opt',
```

Comment:
Here's a hardcoded path to your own wasm installation.

Reply:
Yeah, we don't have a good way to do this actually. I guess now is as good a time as any to fix that :)

Reply:
Done.